### DISCUSSION TITLE
Detect transparency around Dockbarx popup

### DISCUSSION THREAD

**Discussion by Flashwalker** [yshui/picom#1016] [Q&A] [answered]
Is there any chance to make some how a detection of transparency around the [Dockbarx/Dockx](https://github.com/M7S/dockbarx) popup?
![image](https://user-images.githubusercontent.com/2188215/216286963-8b459790-f36c-4121-a993-36b33147def6.png)

**Comment by absolutelynothelix** [comment]
do you want to disable effects (blur, shadow) for it or effects (blur, shadow) to respect it's shape?

**Comment by Flashwalker** [comment]
Well, actually this popup has no blur or shadow by itself. But you can see that the blur (and shadow) applied by picom is applied to the bounding rectangle around the popup, but not to the actual shape of it. So i meant: respect it's form

**Reply by absolutelynothelix** [reply]
> So i meant: respect it's form

unfortunately, the answer is no. it's already done for windows that are using the x shape extension (see https://github.com/yshui/picom/releases/tag/v10-rc1) but there is nothing we can do for non-rectangular windows that aren't using the x shape extension (afaik, i can't find the source now, but i've read it somewhere in the issues, probably).

**Reply by Flashwalker** [reply]
ok, thanks

**Comment by Flashwalker** [comment, marked as answer]
It was fixed/implemented by https://github.com/M7S/dockbarx/issues/173
